src/index.mjs

```diff
@@ -51,10 +51,10 @@ async function fetch(cookieJars, url, options) {
         (!Number.isSafeInteger(options.follow) || options.follow < 0)
     )
         throw new TypeError("options.follow is not a safe positive integer");
+    options.headers = new Headers(options.headers);
     if (cookies) {
         // copy Headers as well so we don't modify it
         // or, if headers is an object, construct a Headers object from it
```

Comment:
Can you also move these two comments in front of the `new Headers()`. They don't make any sense in the place they currently are. Thanks!